server/controllers/thread.controller.js

```diff
@@ -22,4 +23,109 @@ function get(req, res, next) {
         .catch(next);
 }
 
-export default { get };
+function findFirstUnreadMessageId(thread, allMessages) {
+    let minUnreadCreatedDate;
+    let minUnreadMessageId;
+    let j;
+    for (j = 0; j < allMessages.length; j++) { // eslint-disable-line no-plusplus
+        if (allMessages[j].originalMessageId !== thread.originalMessageId) {
+            continue; // eslint-disable-line no-continue
+        }
+        if (minUnreadMessageId) {
+            if (minUnreadCreatedDate > allMessages[j].createdAt) {
+                minUnreadCreatedDate = allMessages[j].createdAt;
+                minUnreadMessageId = allMessages[j].id;
+            }
+        } else {
+            minUnreadMessageId = allMessages[j].id;
+        }
+    }
+    return minUnreadMessageId;
+}
+
+function findLastReadMessageId(thread, allMessages) {
+    let maxCreatedDate;
+    let maxCreatedMessageId;
+    let j;
+    for (j = 0; j < allMessages.length; j++) { // eslint-disable-line no-plusplus
+        if (allMessages[j].originalMessageId !== thread.originalMessageId) {
+            continue; // eslint-disable-line no-continue
+        }
+        if (maxCreatedMessageId) {
+            if (maxCreatedDate < allMessages[j].readAt) {
+                maxCreatedDate = allMessages[j].readAt;
+                maxCreatedMessageId = allMessages[j].id;
+            }
+        } else {
+            maxCreatedMessageId = allMessages[j].id;
+        }
+    }
+    return maxCreatedMessageId;
+}
+
+function findFirstSubjectText(thread, allMessages) {
+    let minSubjectDate;
+    let minSubject;
+    let j;
+    for (j = 0; j < allMessages.length; j++) { // eslint-disable-line no-plusplus
+        if (allMessages[j].originalMessageId !== thread.originalMessageId) {
+            continue;  // eslint-disable-line no-continue
+        }
+        if (!minSubjectDate || minSubjectDate > allMessages[j].createdAt) {
+            minSubjectDate = allMessages[j].createdAt;
+            minSubject = allMessages[j].subject;
+        }
+    }
+    return minSubject;
+}
+
+function list(req, res, next) {
+    const queryObject = {
+        raw: true,
+        where: {},
+    };
+
+    if (req.query.limit) {
+        queryObject.limit = req.query.limit;
+    }
+
+    if (req.query.offset) {
+        queryObject.limit = req.query.limit;
+    }
+
+    const senders = [Sequelize.fn('ARRAY_AGG', Sequelize.fn('DISTINCT', Sequelize.col('from'))), 'senders'];
```

Comment:
I wrote this wrong in the ticket, can we change `senders` to `from` to be consistent with a message object?